Improvements (chapters 1 to 12)

tfrommen opened the pull request:
Besides minimal changes regarding a few `@param` PHPDoc comments in order to use the same format, this pull request contains several improvements for the reguloar twelve chapters:

* Add a bunch of full stops to comments and task descriptions.
* Rephrase sentences to adapt their appearance (e.g., 'Shows all...' -> 'Show all...').
* Add trailing newlines to all files.
* Improve a few comments/task descriptions.
* Don't use `require_once` and `include` as functions, they are language constructs.

thaJeztah commented on 01 - Navigation/01 - Navigate to something you know/Code/JetBrains/Customers/Customer.php:
```diff
@@ -20,7 +20,7 @@ function __construct($name, $age)
 
     /**
-     * @param int $age
+     * @param $age int
```
I think these changes are incorrect; according to the docs, *type* should be *before* the name of the param; http://www.phpdoc.org/docs/latest/references/phpdoc/tags/param.html

tfrommen replied:
I aggree, and I always write `@param` PHPDoc comments à la _type, variable name, description_. However, all other comments throughout the workshop materials are in the manner stated in my commit.

thaJeztah replied:
Perhaps it's best to wait for @maartenba to respond.

If you intend to add the missing newlines to all files, I think it's easier to do so with some simple scripting, like `sed` (for an example, see: http://unix.stackexchange.com/questions/31947/how-to-add-a-newline-to-the-end-of-a-file).

thaJeztah replied:
(Oops, clicked the wrong "add a line note" button.. should've clicked the one below)

thaJeztah commented on 01 - Navigation/01 - Navigate to something you know/Code/JetBrains/Customers/Customer.php:
```diff
@@ -57,4 +57,4 @@ public function getName()
     }
-} 
\ No newline at end of file
+}
```
You seem to have removed newlines at the end of several files. There should be a newline (see, for example; http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline)

tfrommen replied:
Well, interesting that the absence of newlines in **most** of the files did not seem to be of interest to anyone. When adapting a few files to the rest, however, there you come complaining. :)

Again, I'm on your side here&mdash;I just thought it a bit too proactive to change almost any file.

thaJeztah replied:
I didn't look into the other/existing files, just looked at the changes in this PR. Although I agree with being "consistent", I'm not sure introducing (what I see as) errors to be consistent, is the best approach.

Please, don't see this as a personal attack, I really appreciate your efforts. I try to be honest when reviewing code. Also, I'm not affiliated with JetBrains, so I don't have *any* say in this.

tfrommen replied:
I did not take this personal. Sorry if I made it look this way. Having rethought this, I (even more) agree with you in that it would be better to introduce several changes to make things consistent **and** right, compared to just taking the easier route and adapt a small portion to what's already there. Now I'm thinking about improving my commits, but I am unsure on this...

thaJeztah commented on 01 - Navigation/01 - Navigate to something you know/1.1. Project Tool Window.txt:
```diff
@@ -1,10 +1,10 @@
 Project Tool Window
 
-Shows all files in project.
+Show all files in project.
```
I think this should be "Shows" as it describes what the "Project Tool Window" does.

Read it as "(The) Project Tool Window .. *shows* all files in the project"

tfrommen replied:
Please read the rest of the _original_ files. I just rephrased a few to have all written in the same way. Who or what the described thing does is up to the reader, I'd say.

maartenba commented:
Good changes so far!

tfrommen commented:
Hi Maarten,

I was just about to change what I already discussed with @thaJeztah&mdash;newlines and the end of each file, _correct_ order of `@param` PHPDoc comments. That's when I saw your comment...

What do you think about this?

maartenba commented:
Sounds like a plan, I like the idea

thaJeztah commented on 05 - Refactoring/05 - Extract Field.php:
```diff
@@ -14,7 +14,7 @@ class Greeter {
         // 1. Extract the string 'Hello' into a field named $_greeting so we are able to customize the greeting.
         //    Use the "Initialize in class constructor" option to generate a constructor.
-        //    Make it a protected field.
+        //    Make it a protected field..
```
Oops, double `..`

tfrommen replied:
Fixed that.

thaJeztah commented:
Lot better, thanks!

Two remarks;

- You (inadvertently) added a `Thumbs.db`
- There's something weird going on with the PNGs; I cannot get them to show. I cloned your repo locally and the PNGs seem to be broken somehow.

tfrommen commented:
@thaJeztah The `Thumbs.db` file has been there all the time. I deleted it now, though.

I did not see any binary files using PhpStorm's built-in git features. I just reverted any changes to images as well as the icon file. Thanks for that.

thaJeztah commented:
@tfrommen thanks; weird though, if I view the files in your latest commit, GitHub won't show the images:
https://github.com/tfrommen/phpstorm-workshop/commit/f80e755bf9c5bb46194a8d15e5aa3e14828be272#diff-787d37fecfdeb310bccbbf2bc75487f2 (It *does* show them individually [link](https://github.com/tfrommen/phpstorm-workshop/blob/f80e755bf9c5bb46194a8d15e5aa3e14828be272/99%20-%20Miscellaneous/Refactoring/images/asterisk.png))

Viewing the original commit, GitHub also shows them correctly; https://github.com/tfrommen/phpstorm-workshop/commit/82a3b9c0cabb79ff0bb0173bd036cd43191b31f4#diff-787d37fecfdeb310bccbbf2bc75487f2

So, not really sure what's wrong there.. weird

maartenba commented:
Looks good to me, shall I merge it in?

thaJeztah commented:
@maartenba it looks good, but if you could check the PNGs just to be sure nothing's wrong with them (to confirm it's just a GitHub quirk).

thaJeztah commented:
Thanks @tfrommen and bedankt, Maarten :)

maartenba commented:
Thanks to both of you!
